rfcs/20220307-monitoring_across_the_stack.md

```diff
@@ -0,0 +1,353 @@
+[^3]: This could be wrong but I couldn't find any other way to do it that didn't require a script that made HTTP requests. Even the grafana-cli isn't capable of re-evaluating the provisioning scripts.
+
+# Appendices
+
+## Appendix A: Relevant documentation
+
+Below are links to relevant documentation in Prometheus and Grafana. Reviewing these is not necessary for reviewing the RFC. I've included them to help implementers have a good starting point for finding things as sometimes even knowing what terms to search for can be a hurdle.
+
+* Prometheus rules unit tests: https://prometheus.io/docs/prometheus/latest/configuration/unit_testing_rules/
+* Prometheus metric types: https://prometheus.io/docs/concepts/metric_types/
+* Prometheus anomaly detection: https://prometheus.io/blog/2015/06/18/practical-anomaly-detection/
+* Prometheus query functions: https://prometheus.io/docs/prometheus/latest/querying/functions/
+* Prometheus range vectors: https://prometheus.io/docs/prometheus/latest/querying/basics/#range-vector-selectors
+* Prometheus naming best practices: https://prometheus.io/docs/practices/naming/
+* Prometheus official and unofficial client libraries: https://prometheus.io/docs/instrumenting/clientlibs/
+* Adding a datasource to Grafana: https://grafana.com/docs/grafana/latest/datasources/add-a-data-source/
+* Grafana dashboard best practices: https://grafana.com/docs/grafana/latest/best-practices/best-practices-for-creating-dashboards/
+* Histograms and heatmaps in Grafana: https://grafana.com/docs/grafana/latest/basics/intro-histograms/
+* Grafana glossary (super helpful!): https://grafana.com/docs/grafana/latest/basics/glossary/
+
+## Appendix B: Alternative technologies
+
+The following alternatives were considered before landing on Prometheus and Grafana.
+
+### Graphite with Grafana
+
+[Graphite](https://graphiteapp.org/) is very easy to put metrics into and integrates smoothly with Grafana. However, for every bit as easy as Graphite is to put metrics into, it is also that difficult to create queries against. Graphite does not hold your hand at all. It supports anomaly detection through [additional services](https://github.com/etsy/skyline), but they're not actively developed and configuring the anomaly detections requires relatively deep knowledge about statistics that I don't think our team is necessarily prepared to invest time into (nor do we have the existing knowledge here as far as I know).
+
+Graphite is also older than Prometheus and while it works really well, there's a lot less current information about how to deploy and configure it. There's also almost no support for languages outside of the core langauges supported by the project including poor JavaScript support (which would make monitoring our Nuxt service more difficult).
+
+### CloudWatch
```

Comment:
I agree with all these points and would love to NOT use CloudWatch where possible :)